Polars 0.52 fails to build with Lazy

### Checks

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

```toml
[dependencies]
polars = { version = "0.52", features = ["lazy"] }
```

### Log output

```shell
error[E0433]: failed to resolve: use of undeclared type `DataType`
  --> /home/eric/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/polars-expr-0.52.0/src/dispatch/datetime.rs:76:9
   |
76 |         DataType::Datetime(_, _) => s
   |         ^^^^^^^^ use of undeclared type `DataType`
   |
   = note: enum `crate::reduce::var_std::DataType` exists but is inaccessible
help: consider importing this enum
   |
 3 + use polars_core::prelude::DataType;
```

### Issue description

When running Polars 0.52 (Rust) with the `Lazy`, it fails to build with a `failed to resolve: use of undeclared type 'DataType'` error. No code necessary, simply fails to build with a Hello World crate.

You can see a CI failing example here: https://github.com/EricFecteau/df-interchange/actions/runs/19051322031/job/54412020896?pr=15#step:9:330

### Expected behavior

Can build with Lazy.

### Installed versions

<details>

```toml
[dependencies]
polars = { version = "0.52", features = ["lazy"] }
```

</details>

--- a/crates/polars-expr/src/dispatch/datetime.rs
+++ b/crates/polars-expr/src/dispatch/datetime.rs
@@ -1,9 +1,11 @@
 #[cfg(feature = "timezones")]
 use arrow::legacy::time_zone::Tz;
 use polars_core::error::{PolarsResult, polars_bail};
-use polars_core::prelude::{ArithmeticChunked, Column, IntoColumn, LogicalType, TimeUnit};
+use polars_core::prelude::{
+    ArithmeticChunked, Column, DataType, IntoColumn, LogicalType, TimeUnit,
+};
 #[cfg(feature = "timezones")]
-use polars_core::prelude::{DataType, NonExistent, StringChunked, TimeZone};
+use polars_core::prelude::{NonExistent, StringChunked, TimeZone};
 use polars_time::prelude::*;
 use polars_time::replace_datetime;
 use polars_time::series::TemporalMethods;

